File: problems/1491-average-salary-excluding-the-minimum-and-maximum-salary/solution.java
class Solution {
    public double average(int[] salary) {
        int minSalary = salary[0];
        int maxSalary = salary[0];
        double sum = 0;

        for (int s : salary) {
            minSalary = Math.min(minSalary, s);
            maxSalary = Math.max(maxSalary, s);
            sum += s;
        }

        sum -= minSalary + maxSalary;
        return sum / (salary.length - 2);
    }
}

File: problems/1491-average-salary-excluding-the-minimum-and-maximum-salary/solution.c
double average(int* salary, int salarySize) {
    int minSalary = salary[0];
    int maxSalary = salary[0];
    double sum = 0;

    for (int i = 0; i < salarySize; i++) {
        if (salary[i] < minSalary) {
            minSalary = salary[i];
        }
        if (salary[i] > maxSalary) {
            maxSalary = salary[i];
        }
        sum += salary[i];
    }

    sum -= (minSalary + maxSalary);
    return sum / (salarySize - 2);
}